### PR TITLE
Ensure BCF snapshot column works again

### PR DESCRIPTION
**Pull request by wielinde**
Fixes the BCF schema patch
https://community.openproject.com/wp/32514

### CODE REVIEW

File: modules/bim/spec/features/bcf/bcf_snapshot_column_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                      member_in_project: project,
+                      member_with_permissions: permissions
+  end
+
+
+  before do
+    login_as(user)
+  end
+
+  let!(:query) do
```
Comment (HDinger) [inline modules/bim/spec/features/bcf/bcf_snapshot_column_spec.rb:14]: Just for code quality issues: Please move this definition to other ones above the `before` block. :)

File: frontend/src/app/modules/bim/bcf/fields/display/bcf-thumbnail-field.module.ts [hunk 2]
```diff
@@ -27,17 +27,17 @@
...
+import {BcfPathHelperService} from "core-app/modules/bim/bcf/helper/bcf-path-helper.service";
 
 export class BcfThumbnailDisplayField extends DisplayField {
-  @InjectField() pathHelper:PathHelperService;
+  @InjectField() bcfPathHelper:BcfPathHelperService;
 
   public render(element:HTMLElement, displayText:string):void {
-    if (_.get(this, 'resource.bcf.viewpoints[0]')) {
-      let vp = this.resource.bcf.viewpoints[0];
+    if (_.get(this, 'resource.bcfViewpoints[0]')) {
```
Comment (HDinger) [inline frontend/src/app/modules/bim/bcf/fields/display/bcf-thumbnail-field.module.ts:37]: I don't think that we still need the way with the `_get` method. I guess `resource.bcfViewpoints` should already deliver everything we need, similar to how it is done here: https://github.com/opf/openproject/blob/dev/frontend/src/app/components/wp-card-view/wp-single-card/wp-single-card.component.ts#L120

**Review by HDinger — COMMENTED** [review]
The thumbnails are displayed again. I have a few styling tissues that do not necessarily have to be covered in this PR.:

* When clicking on a thumbnail, there is an outline set and the row changes its height a bit. This looks strange.
* The images should be centered in the cell.
* I guess the images could use some padding. When we habe multiple large thumbnails below each other, it looks squeezed:
Bildschirmfoto 2020-03-12 um 10 01 45

One other thing: When I am not logged in, I cannot see any thumbnails. Was this done on purpose? 
Please also have a look at the failing test.

File: frontend/src/app/modules/bim/bcf/fields/display/bcf-thumbnail-field.module.ts (hunk 2 above)
Comment (wielinde) [reply frontend/src/app/modules/bim/bcf/fields/display/bcf-thumbnail-field.module.ts:37]: Hmm, I think I don't see the solution that you are seeing. `resource.bcfViewpoints` can be `null`. Let's talk about that tomorrow.

**Comment by wielinde** [comment]
@HDinger Thank you for your review.

- I don't understand why the spec failed. Works locally. Maybe concerned with some weird caching BS. 
- Every cell in the table has an outline on focus. I hacked a bit to get rid of it for the thumbnails.
- I preferred align left, as we are in the context of a table and text would align left, too. No strong opinion here.
- I didn't add padding on purpose to use the maximum of space to show the images. I don't want the line height to change as this would mess with the Gantt chart. So adding a padding must make the thumbnail smaller... and thus you would see less. I don't like. They look squeezed? Hmm, maybe I am already blind of seeing it too often.

File: frontend/src/app/modules/bim/bcf/fields/display/bcf-thumbnail-field.module.ts (hunk 2 above)
Comment (wielinde) [reply frontend/src/app/modules/bim/bcf/fields/display/bcf-thumbnail-field.module.ts:37]: I changed those parts as discussed in the meeting.

**Comment by wielinde** [comment]
I close this PR in favour of #8153 which tries to trick Travis

**Pull request closed without merging**